### PR TITLE
[Bug][Datafusion] fix TaskContext session_config bug

### PR DESCRIPTION
**Pull request by gaojun2048**
closes https://github.com/apache/arrow-datafusion/issues/2069

### CODE REVIEW

File: ballista-examples/examples/test_sql.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    // register csv file with the execution context
+    ctx.register_csv(
+        "aggregate_test_100",
+        &format!("{}/csv/aggregate_test_100.csv", testdata),
+        CsvReadOptions::new(),
+    )
+    .await?;
+
+    // test udf
+    let df = ctx.sql("select count(1) from aggregate_test_100").await?;
```
Comment (alamb) [inline ballista-examples/examples/test_sql.rs:42]: 💯  for test

File: datafusion/src/execution/context.rs [hunk 2]
```diff
@@ -1340,27 +1340,31 @@ impl TaskContext {
...
-                        props.get(REPARTITION_WINDOWS).unwrap().parse().unwrap(),
-                    )
-                    .with_parquet_pruning(
-                        props.get(PARQUET_PRUNING).unwrap().parse().unwrap(),
-                    )
+                if props.is_empty() {
+                    session_config
+                } else {
+                    session_config
+                        .with_batch_size(props.get(BATCH_SIZE).unwrap().parse().unwrap())
```
Comment (alamb) [inline datafusion/src/execution/context.rs:1347]: FYI @mingmwang

**Review by alamb — APPROVED** [review]
LGTM

**Comment by alamb** [comment]
None of the CI checks ran on this branch @ https://github.com/apache/arrow-datafusion/pull/2070/commits/ccf67091c8dd5e2045802241c411ce56fa021b5e (probably because github was having issuesearlier today ) so I merged `apache/master` into this branch in an attempt to start the checks

File: ballista-examples/examples/test_sql.rs (hunk 1 above)
Comment (liukun4515) [reply ballista-examples/examples/test_sql.rs:42]: @alamb 
Do we have the UT or IT for the ballista side?
I think this `test_sql.rs` file just adds an example for the ballista.

Comment (liukun4515) [reply ballista-examples/examples/test_sql.rs:42]: Maybe we can add this test for the UT or it IT framework.

Comment (gaojun2048) [reply ballista-examples/examples/test_sql.rs:42]: @liukun4515  I think we need a ballista-integration-test cargo.

**Comment by mingmwang** [comment]
My bad. The properties in the TaskContext is not populated in this PR.  

The fix LGTM.

File: ballista-examples/examples/test_sql.rs (hunk 1 above)
Comment (gaojun2048) [reply ballista-examples/examples/test_sql.rs:42]: @liukun4515  How about add a ballista-integration-test cargo?

**Comment by alamb** [comment]
Thanks @gaojun2048

File: ballista-examples/examples/test_sql.rs (hunk 1 above)
Comment (alamb) [reply ballista-examples/examples/test_sql.rs:42]: I think https://github.com/apache/arrow-datafusion/issues/321 tracks some of this -- I will make a note

Comment (liukun4515) [reply ballista-examples/examples/test_sql.rs:42]: > about add a ballista-integration-

agree with the suggestion.
we can discuss this in this issue https://github.com/apache/arrow-datafusion/issues/321.

@gaojun2048  you can provide some suggestions or proposals.
You can point out some lack of ballista about test and record them.
We can make a plan to address them in Q2.

**Pull request merged**